lib/elixir/lib/calendar/iso.ex

```diff
@@ -588,10 +628,10 @@ defmodule Calendar.ISO do
 
   """
   @doc since: "1.4.0"
-  @spec day_of_week(year, month, day) :: 1..7
+  @spec day_of_week(year, month, day) :: day_of_week
   @impl true
   def day_of_week(year, month, day)
-      when is_integer(year) and is_integer(month) and is_integer(day) do
+      when is_year(year) and is_month(month) and is_day(day) do
```

Comment:
Let's remove the guards here since we already check them on `date_to_iso_days` (if not, we should).